src/components/form/HdForm.vue

```diff
@@ -31,8 +31,22 @@ export default {
       fields: [],
     };
   },
+  computed: {
+    fieldsValues() {
+      return this.fields.map(field => field.value);
+    },
+  },
   created() {
     this.initialFormData = _cloneDeep(this.getFormData());
+
+    if (this.$listeners.hasChanged) {
+      this.$watch('fieldsValues', () => {
```

Comment:
I don't know if is relevant but maybe we need to make sure to `unwatch` this before the component is destroyed: 

``` javascript
var unwatch = vm.$watch('a', cb)
// later, teardown the watcher
unwatch()
```

I don't know if Vue handles this automatically but maybe it worth testing, otherwise we might face some memory leaks